File: Algorithms/20. Valid Parentheses/golang/solution.go
func isValid(s string) bool {
	mapping := map[rune]rune{')': '(', ']': '[', '}': '{'}
	var stack []rune

	for _, char := range s {
		if char == '(' || char == '[' || char == '{' {
			stack = append(stack, char)
		} else if len(stack) > 0 && mapping[char] == stack[len(stack)-1] {
			stack = stack[:len(stack)-1]
		} else {
			return false
		}
	}

	return len(stack) == 0
}

File: Algorithms/20. Valid Parentheses/swift/solution.swift
class Solution {
    func isValid(_ s: String) -> Bool {
        var stack: [Character] = []
        for char in s {
            if char == "(" || char == "[" || char == "{" {
                stack.append(char)
            } else if let lastStack = stack.last {
                if lastStack == "(", char == ")" {
                    _ = stack.popLast()
                } else if lastStack == "[", char == "]" {
                    _ = stack.popLast()
                } else if lastStack == "{", char == "}" {
                    _ = stack.popLast()
                } else {
                    return false
                }
            } else {
                return false
            }
        }

        return stack.isEmpty
    }
}